ts/demo-functions/src/kubeval.ts

```diff
@@ -0,0 +1,164 @@
+import {
+  Configs,
+  KubernetesObject,
+  kubernetesObjectResult,
+  Result,
+} from 'kpt-functions';
+import { ChildProcess, spawn } from 'child_process';
+import { Writable } from 'stream';
+
+const SCHEMA_LOCATION = 'schema_location';
+const ADDITIONAL_SCHEMA_LOCATIONS = 'additional_schema_locations';
+const IGNORE_MISSING_SCHEMAS = 'ignore_missing_schemas';
+const SKIP_KINDS = 'skip_kinds';
+const STRICT = 'strict';
+
+type Feedback = FeedbackItem[];
+
+interface FeedbackItem {
+  filename: string;
+  kind: string;
+  status: 'valid' | 'invalid';
+  errors: string[];
+}
+
+export async function kubeval(configs: Configs): Promise<void> {
+  const schemaLocation = configs.getFunctionConfigValue(SCHEMA_LOCATION);
+  const additionalSchemaLocationsStr = configs.getFunctionConfigValue(
+    ADDITIONAL_SCHEMA_LOCATIONS
+  );
+  const additionalSchemaLocations = additionalSchemaLocationsStr
+    ? additionalSchemaLocationsStr.split(',')
+    : [];
+  const ignoreMissingSchemas = JSON.parse(
+    configs.getFunctionConfigValue(IGNORE_MISSING_SCHEMAS) || 'false'
+  );
+  const skipKindsStr = configs.getFunctionConfigValue(SKIP_KINDS);
+  const skipKinds = skipKindsStr ? skipKindsStr.split(',') : [];
+  const strict = JSON.parse(configs.getFunctionConfigValue(STRICT) || 'false');
+
+  const results: Result[] = [];
+
+  for (const object of configs.getAll()) {
+    await runKubeval(
+      object,
+      results,
+      schemaLocation,
+      additionalSchemaLocations,
+      ignoreMissingSchemas,
+      skipKinds,
+      strict
+    );
+  }
+
+  if (results.length > 0) {
+    configs.addResults(...results);
+  }
+}
+
+async function runKubeval(
+  object: KubernetesObject,
+  results: Result[],
+  schemaLocation?: string,
+  additionalSchemaLocations?: string[],
+  ignoreMissingSchemas?: boolean,
+  skipKinds?: string[],
+  strict?: boolean
+): Promise<void> {
+  const args = ['--output', 'json'];
+
+  if (schemaLocation) {
```

Comment:
Let's move these if statements to the main kubeval function and pass an args?: string[] argument to runKubeval. Also I think you don't need if statements for the arguments you set defaults for.